scripts/base/frameworks/software/main.zeek

```diff
@@ -514,17 +535,8 @@ function found(id: conn_id, info: Info): bool
 		return F;
 		}
 
-	if ( ! info?$version )
-		{
-		local sw = parse(info$unparsed_version);
-		info$unparsed_version = sw$unparsed_version;
-		info$name = sw$name;
-		info$version = sw$version;
-		}
-
```

Comment:
I suspect that it's the relocation of the parsing logic that leads CI to fail with lots of `expression error in /zeek/scripts/policy/protocols/http/detect-webapps.zeek, line 44: field value missing (HTTP::info$name)`. I'm tempted to just leave it where it is and just do the caching there, does it really make a lot of a difference?

Reply:
ah, I didn't notice that other scripts used Software::register.  I can easily fix that, just need an intermediary event. to keep the current API the same.

Moving the parsing and caching to the proxies should make a large difference:

* It completely offloads the parsing to the proxies, so the workers don't have to parse at all.
* It lets you have a small number of per proxy caches instead of a large number of per worker caches.

The cache on the proxies should be mostly free, since references to all the strings will be stored in the tracked software table anyway.  The workers don't have that table, so a cache there would increase memory usage.